crates/cxx-qt-lib-extras/src/core/qlocale.cpp

```diff
@@ -0,0 +1,23 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Nicolas Fella <nicolas.fella@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib-extras/qlocale.h"
+
+namespace rust {
+namespace cxxqtlib1 {
+QLocale fromName(const QString& name)
+{
+    return QLocale(name);
+}
+
+QLocale defaultLocale()
+{
+    return QLocale();
+}
+}
+}
+
```

Comment:
the rust side has a `usize` as it's size, add a check here using `assert_alignment_and_size(QLocale, { ::std::size_t a0; });` to ensure that is true.

There needs to be a combination of static asserts on if it's trivially copyable/assignable/constructible/destructible etc.

If it is not one of these then you need to implement Drop/Clone manually etc (have a look around some of our other types to get ideas :-) )